test/core/util/fuzzing_channel_args.h

```diff
@@ -0,0 +1,44 @@
+// Copyright 2023 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GRPC_TEST_CORE_UTIL_FUZZING_CHANNEL_ARGS_H
+#define GRPC_TEST_CORE_UTIL_FUZZING_CHANNEL_ARGS_H
+
+#include <grpc/support/port_platform.h>
+
+#include "absl/strings/string_view.h"
+
+#include "src/core/lib/channel/channel_args.h"
+#include "src/core/lib/gprpp/ref_counted_ptr.h"
+#include "src/core/lib/resource_quota/resource_quota.h"
+#include "test/core/util/fuzzing_channel_args.pb.h"
+
+namespace grpc_core {
+namespace testing {
+
+// Create ChannelArgs from a fuzzer configuration.
+// resource_quota is an optional input/output argument. If it's not a nullptr,
+// it will be set on the ChannelArgs object. If it is a nullptr, then it will be
+// set to a ResourceQuota object iff the fuzzer configuration has requested that
+// a ResourceQuota be created.
+// The prefix_identifier is a name given to the ResourceQuota if one is created.
+ChannelArgs CreateFuzzingChannelArgs(
+    const grpc::testing::FuzzingChannelArgs& fuzzing_channel_args,
+    absl::string_view prefix_identifier,
+    RefCountedPtr<ResourceQuota>* resource_quota = nullptr);
```

Comment:
let's change this now to be something like:
```
ChannelArgs CreateFuzzingChannelArgs(
    const grpc::testing::FuzzingChannelArgs& fuzzing_channel_args,
    absl::string_view prefix_identifier,
    FuzzingEnvironment fuzzing_environment);
 ```
 with
```
struct FuzzingEnvironment {
  optional<RefCountedPtr<ResourceQuota>> resource_quota;
};
```

we'll want to add more types in the future that get bound into the fuzzers, may as well set things up so it's an O(1) operation when it happens.

Reply:
Good idea, I added something similar. It's fairly clean and has maybe one unnecessary allocation in some cases (probably none when we get designated initializers with C++20 ... someday).

New thought for another improvement: the API fuzzer currently uses the same resource quota in both the client & server channels. It seems valuable to split them up to exercise separate reclamation actions for clients & servers. The logic gets a bit nicer if we make a proto change, but the following could work if we don't want to change the proto.

```
message Action {
    reserved 2;
    oneof type {
        ...
        // resize both client and server quotas.
        uint32 resize_resource_quota = 22;
        // use separate quotas. These are ignored if resize_resource_quota is specified.
        uint32 resize_client_resource_quota = 23;
        uint32 resize_server_resource_quota = 23;
    }
}
```